spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/MergeInto.scala

```diff
@@ -21,13 +21,12 @@ package org.apache.spark.sql.catalyst.plans.logical
 
 import org.apache.spark.sql.catalyst.expressions.Attribute
 import org.apache.spark.sql.catalyst.expressions.Expression
-import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation
 
 case class MergeInto(
     mergeIntoProcessor: MergeIntoParams,
-    targetRelation: DataSourceV2Relation,
+    targetOutput: Seq[Attribute],
```

Comment:
I think this could just be `output` and you wouldn't need to override `def output` below.